swap/peer.go

```diff
@@ -80,12 +82,36 @@ func (sp *Peer) handleEmitChequeMsg(ctx context.Context, msg interface{}) error
 		return fmt.Errorf("Invalid message type, %v", msg)
 	}
 	cheque := chequeMsg.Cheque
+
+	if cheque.Contract != sp.contractAddress {
+		return fmt.Errorf("wrong cheque parameters: expected contract: %s, was: %s", sp.contractAddress, cheque.Contract)
+	}
+
+	// the beneficiary is the owner of the counterparty swap contract
+	err := sp.swap.verifyChequeSig(cheque, sp.beneficiary)
+
+	if err != nil {
+		log.Error("error invalid cheque", "from", sp.ID().String(), "err", err.Error())
+		return err
+	}
+
+	if cheque.Beneficiary != sp.swap.owner.address {
+		return fmt.Errorf("wrong cheque parameters: expected beneficiary: %s, was: %s", sp.swap.owner.address, cheque.Beneficiary)
+	}
+
+	if cheque.Timeout != 0 {
+		return fmt.Errorf("wrong cheque parameters: expected timeout to be 0, was: %d", cheque.Timeout)
+	}
+
+	// TODO: check serial and balance are higher
+
 	// reset balance to zero, TODO: fix
 	sp.swap.resetBalance(sp.ID())
 	// send confirmation
-	err := sp.Send(ctx, &ConfirmMsg{})
+	err = sp.Send(ctx, &ConfirmMsg{})
 	if err != nil {
-		log.Error(fmt.Sprintf("error while sending confirm msg to peer %s: %s", sp.ID().String(), err.Error()))
+		log.Error("error while sending confirm msg to peer", sp.ID().String(), err.Error())
```

Comment:
Nearly...the format is `log.Error(msg, key1, value1, key2, value2,...)`

Thus it would be `log.Error("error while sending confirm msg to peer","peer",sp.ID().String(), "err", err.Error())